common/src/main/java/com/microsoft/identity/common/internal/cache/IAccountCredentialCache.java

```diff
@@ -1,28 +1,116 @@
 package com.microsoft.identity.common.internal.cache;
```

Comment:
I think I'm happy with the state of this PR and, if looks OK to you @oldalton / @addev-ashish I'd like to merge it into #50 

Does the `IAccountCredentialCache` feature everything we need from our caching layer? Does `removeAccount(String, String)` and `removeCredential(Credential)` provide an adequate interface to support logging users out?

Reply:
why don't we do removeAccount(Account)?

Reply:
Can do - @oldalton do you have any preference for providing this method **instead** of `removeAccount(String, String)` versus **in addition to**?

Reply:
I think we need a way to remove a specific item and multiple items, but they should be called differently.